src/gauging.jl

```diff
@@ -70,161 +110,56 @@ function vidal_gauge(
-function vidal_to_symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
-  ψsymm = copy(ψ)
-  ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
-  ψsymm_mts = default_bp_cache(pψψsymm)
-
-  for e in edges(ψsymm)
-    vsrc, vdst = src(e), dst(e)
-    pe = partitionedge(pψψsymm, NamedEdge((vsrc, 1) => (vdst, 1)))
-    root_S = sqrt_diag(bond_tensors[e])
-    setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vsrc]), vsrc)
-    setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vdst]), vdst)
-
-    ψsymm_mts[pe], ψsymm_mts[reverse(pe)] = copy(ITensor[dense(bond_tensors[e])]),
-    copy(ITensor[dense(bond_tensors[e])])
+  if isnothing(cache!)
+    cache! = Ref(default_norm_cache(ψ))
   end
-
-  ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
-
-  return ψsymm, pψψsymm, ψsymm_mts
-end
-
-"""Put an ITensorNetwork into the symmetric gauge and also return the message tensors (which are the diagonal bond matrices from the Vidal Gauge)"""
-function symmetric_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  svd_kwargs...,
-)
-  ψ_vidal, bond_tensors = vidal_gauge(
-    ψ;
-    eigen_message_tensor_cutoff,
-    regularization,
-    niters,
-    target_canonicalness,
-    svd_kwargs...,
-  )
-
-  return vidal_to_symmetric_gauge(ψ_vidal, bond_tensors)
+  cache![] = update(cache![]; cache_update_kwargs...)
+  return vidalitensornetwork_preserve_cache(ψ; bp_cache=cache![], kwargs...)
 end
 
-"""Transform from the Symmetric Gauge (message tensors) to the Vidal Gauge (bond tensors)"""
-function symmetric_to_vidal_gauge(
-  ψ::ITensorNetwork,
-  pψψ::PartitionedGraph,
-  mts;
-  regularization=10 * eps(real(scalartype(ψ))),
-)
-  bond_tensors = DataGraph{vertextype(ψ),Nothing,ITensor}(underlying_graph(ψ))
-
-  ψ_vidal = copy(ψ)
-
-  for e in edges(ψ)
-    vsrc, vdst = src(e), dst(e)
-    pe = partitionedge(pψψ, NamedEdge((vsrc, 1) => (vdst, 1)))
-    bond_tensors[e], bond_tensors[reverse(e)] = only(mts[pe]), only(mts[pe])
-    invroot_S = invsqrt_diag(map_diag(x -> x + regularization, bond_tensors[e]))
-    setindex_preserve_graph!(ψ_vidal, noprime(invroot_S * ψ_vidal[vsrc]), vsrc)
-    setindex_preserve_graph!(ψ_vidal, noprime(invroot_S * ψ_vidal[vdst]), vdst)
-  end
-
-  return ψ_vidal, bond_tensors
+function update(ψv::VidalITensorNetwork; kwargs...)
+  return VidalITensorNetwork(ITensorNetwork(ψv); kwargs...)
 end
 
 """Function to measure the 'isometries' of a state in the Vidal Gauge"""
-function vidal_itn_isometries(
-  ψ::ITensorNetwork,
-  bond_tensors::DataGraph;
-  edges=vcat(NamedGraphs.edges(ψ), reverse.(NamedGraphs.edges(ψ))),
+function vidal_gauge_isometries(
+  ψv::VidalITensorNetwork;
+  edges=vcat(NamedGraphs.edges(ψv), reverse.(NamedGraphs.edges(ψv))),
```

Comment:
```suggestion
  ψv::VidalITensorNetwork,
  edges=vcat(NamedGraphs.edges(ψv), reverse.(NamedGraphs.edges(ψv))),
```
I think it makes more sense for the edges to be passed as a positional argument.

Also, could we refactor this in terms of a core function `vidal_gauge_isometry(ψ::VidalITensorNetwork, edge)`, and then `vidal_gauge_isometries` loops over that function?

Another point is that I find it confusing when you call the variable `ψv`, at first I thought `v` was referring to a vertex, but I see now it is probably a shorthand for `vidal`. I think we should either just call it `ψ` when it is unambiguous what it refers to, or `ψ_vidal` if it seems necessary to be more explicit.

Reply:
Seems reasonable! I have refactored this way and used `ψ::VidalITensorNetwork` aside from in a few functions where I need to differentiate and use `ψ_vidal`